pkg/helm/watcher/controller/helm_watcher.go

```diff
@@ -0,0 +1,103 @@
+package controller
+
+import (
+	"context"
+
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	"github.com/go-logr/logr"
+	"github.com/helm/helm/pkg/chartutil"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/weaveworks/weave-gitops/pkg/helm"
+	"github.com/weaveworks/weave-gitops/pkg/helm/watcher/cache"
+)
+
+// HelmWatcherReconciler runs the `reconcile` loop for the watcher.
+type HelmWatcherReconciler struct {
+	client.Client
+	Cache       cache.Cache
+	RepoManager helm.HelmRepoManager
+}
+
+// +kubebuilder:rbac:groups=helm.watcher,resources=helmrepositories,verbs=get;list;watch
+// +kubebuilder:rbac:groups=helm.watcher,resources=helmrepositories/status,verbs=get
+
+// Reconcile is either called when there is a new HelmRepository or, when there is an update to a HelmRepository.
+// Because the watcher watches all helmrepositories, it will update data for all of them.
+func (r *HelmWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
does this mean we scan every HelmRepository resource now? Before it was just the 1 that was configured

Reply:
We should be...

There's an argument to be made to only scan ones with specific labels, but that's probably an optimisation :-)

Reply:
Yep, we scan everything. This will be a good stepping stone to work from later when profiles supports multiple HelmRepositories.

Reply:
The what from the where why what? :D

Reply:
Oh, no that shit is still needed otherwise the GetProfiles commands will not find the helmRepository. That's a different thing separate from this implementation.

Reply:
Sadly, that stuff will still only work with a single HelmRepository until fixed.

Reply:
Opened https://github.com/weaveworks/weave-gitops/issues/1306 to track to switch to tracking and using multiple HelmRepositories.